### ISSUE TITLE
testCheck.py delivers command line options as a single string in argv[1]

### ISSUE THREAD

**Issue by tjayrush** [Great-Hill-Corporation/quickBlocks#232]
I think the testCheck.py script delivers the command line options incorrectly. Here's what I did.

All my code is c++, which means that every one of them starts with 

    int main(int argc, const char *argv[]) {
       // ... whatever
    }

If I change the code to look like this:

    int main(int argc, const char *argv[]) {
       cout << "argc: " << argc << " ";
       for (int i=1;i<argc;i++) {  // start with '1' since '0' is program name
            cout << "[" << i << ": " << argv[i] << "] ";
       }
       cout << "\n";
       // ... whatever
    }

I would expect that if I ran the code from the command line, with this command:

    ethName --param1 --param2 value

it would print out

    argc: 4 [1: --param1] [2: --param2] [3: value]

and it does if I run the tool from the command line.

If I run the test case through ./bin/testCheck.py, however, I get this:

    argc: 2 [1: --param1 --param2 value]

Somehow, I think testCheck.py is delivering all the parameters in a single command line argument, and this makes sense if I look at the CMakeLists.txt files because they use ${ARGV1}

I need the 'run_test' code to deliver each command line option as a separate argument.

**Comment by tjayrush** [comment]
Never mind. I figured it out.

**Comment by tjayrush** [comment]
I solved this by using ${ARGN} instead of ${ARGV1} everywhere it occurred (in all CMakeLists.txt files). I then had to change the run_test lines in all CMakeLists.txt file by separating out the parameters, so this:

    run_test("ethName_invalid_param_8" "--sources 0x17 rob other")

became this:

    run_test("ethName_invalid_param_8" "--sources" " 0x17" " rob" " other")

The only trouble I ran into was that it would only work if I put a leading space inside the quotes for the second and subsequent parameters. It would be better if I didn't have to include the leading space, but this change fixed about 10 different previously broken test cases.
